sympy/simplify/hyperexpand.py

```diff
@@ -2304,7 +2306,7 @@ def do_slater(an, bm, ap, bq, z, zfinal):
                 # Now sum the finitely many residues:
                 # XXX This speeds up some cases - is it a good idea?
                 integrand = expand_func(integrand)
-                for r in range(lu):
+                for r in range(int(lu)):
```

Comment:
I never before looked into the Python docs of ``round``. It says that the result is a floating point number. So perhaps this should be ``int(round(lu))``? There is also ``lu.round()`` but it looks like it would also return a float.